### PR TITLE
Idm milestone

### PR DESCRIPTION
**Pull request by markg58**
Jira issue? CLOUD-1078
Release 6.5.0 backport required? no
6.5.0 doc changes needed? no
7.0.0 doc changes needed? no
Required README updates made? no

### CODE REVIEW

File: cicd/forgeops-tests/lib/utils/kubectl.py [hunk 1]
```diff
@@ -62,4 +62,4 @@ def __run_cmd_process(cmd):
     response = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
     stdout, stderr = response.communicate()
     assert response.returncode == 0, ' Unexpected return code from Popen() ' + stderr
-    return (stdout.split('\n'), stderr.split('\n'))
+    return stdout.split('\n'), stderr.split('\n')
```
Comment (wstrange) [inline cicd/forgeops-tests/lib/utils/kubectl.py]: I do wonder if long term parsing kubectl commands is going to be harder than using the k8s py libarary. But we can cross that bridge later...

File: cicd/forgeops-tests/config/6.5.0-manifest.txt [hunk 2]
```diff
@@ -21,10 +21,10 @@ commons_version = 24.0.1
...
-date = 20190123140439
+version = 7.0.0-M2
+revision = 8a62b3d
+date = 20190213220443
 jdk_version = openjdk version "11.0.1" 2018-10-16
 jre_version = OpenJDK Runtime Environment (build 11.0.1+13-Debian-2bpo91)
 jdk_full_version = OpenJDK 64-Bit Server VM (build 11.0.1+13-Debian-2bpo91, mixed mode)
-commons_version = 25.0.0-alpha-14
\ No newline at end of file
+commons_version = 25.0.0-alpha-21
```
Comment (wstrange) [inline cicd/forgeops-tests/config/6.5.0-manifest.txt]: this is really a 7.0 manifest, right?

File: cicd/forgeops-tests/lib/utils/kubectl.py (hunk 1 above)
Comment (csovant) [reply cicd/forgeops-tests/lib/utils/kubectl.py]: There are pro and cons to use the k8s py lib or other libs (like the helm) one.
- pro: you don't have to manipulate output this is manage by the lib, but the cost is not huge from my point of view
- cons: you can't display equivalent command line, this is really useful while debugging tests/deployments => this also allows us to generate shell scripts from the tests, we had that with PyForge and this was really useful for dev to diagnose/investigate an issue

File: cicd/forgeops-tests/config/6.5.0-manifest.txt (hunk 2 above)
Comment (markg58) [reply cicd/forgeops-tests/config/6.5.0-manifest.txt]: This file will be going away in the future, as we decided the best we can do right now is just to report on the various items and not attempt to validate against some "authoritative source" that doesn't exists.

**Comment by markg58** [comment]
@dgoldssfo - idm_milestone is the branch from which I created the code changes.  JIRA # was included "Jira issue? CLOUD-1078", commit message is "first idm metadata test".  Which actually will be change in near future not to be tests, but just metadata reported as part of the test report.

File: cicd/forgeops-tests/lib/utils/idm_pod.py [hunk 3]
```diff
@@ -51,9 +51,28 @@ def is_expected_version(self):
...
+        stdout, stderr = kubectl.exec(namespace, ' '.join([self.name, '-c openidm', '-- ls', test_jar_filepath]))
+
+        logger.debug('Check commons version for: ' +  self.name)
+        assert stderr[0].strip() == '', 'Did not find ' + test_jar_filepath
+        return True
+
+    def is_expected_jdk(self, namespace):
+        """
+        Return True if jdk is as expected, otherwise assert.
+        :param namespace The kubernetes namespace.
```
Comment (andrewvinall) [inline cicd/forgeops-tests/lib/utils/idm_pod.py]: `:param namespace: The kubernetes namespace.`

File: cicd/forgeops-tests/lib/utils/idm_pod.py [hunk 4]
```diff
@@ -51,9 +51,28 @@ def is_expected_version(self):
...
 
-    def is_expected_commons_version(self):
+    def is_expected_commons_version(self, namespace):
         """
-        Return True if config jar is the expected version
-        :return: True if jar filename contains expected version
+        Return true if the commons version is as expected, otherwise return assert.
+        :param namespace The kubernetes namespace.
+        This check inspects a sample commons .jar to see what version is in use.
+        :return: True is the commons version is as expected.
```
[the remark this answers is not in the document]
Comment (markg58) [reply cicd/forgeops-tests/lib/utils/idm_pod.py]: Sorry, missed these comments.  Will clean up in another PR.

**Pull request merged**